src/pages/accounts/SignUpPage.tsx

```diff
@@ -95,93 +90,72 @@ function SignUpPage() {
-					<InputBox
-						type="text"
-						value={confirmEmail}
-						onChange={(e) => {
-							if (compareEmail(email, confirmEmail))
-								setEmailErr(false);
-							setConfirmEmail(e.target.value);
-						}}
-						onBlur={() => {
-							if (!compareEmail(email, confirmEmail))
-								setEmailErr(true);
-						}}
-					/>
-				</InputContainer>
-
-				<InputContainer>
-					<InputLabel>
-						<label className="Password-label">Password *</label>
-					</InputLabel>
-					<InputBox
-						type="password"
-						value={userPass}
-						onChange={(e) => {
-							if (validPass(e.target.value)) setPassErr(false);
-							setUserPass(e.target.value);
-						}}
-					/>
-				</InputContainer>
-
-				<InputContainer>
-					<InputLabel>
-						<label className="Password-label">
-							Confirm Password *
-						</label>
-					</InputLabel>
-					<InputBox
-						type="password"
-						value={confirmUserPass}
-						onChange={(e) => {
-							if (comparePass(userPass, confirmUserPass))
-								setPassErr(false);
-							setConfirmUserPass(e.target.value);
-						}}
-					/>
-				</InputContainer>
-
-				<Links>
-					<a
-						href="https://reactjs.org"
-						target="_blank"
-						rel="noopener noreferrer"
-					>
-						Already have an account? Log in here.
-					</a>
-				</Links>
-				<SignUpButton
-					type="button"
-					disabled={emailErr || passErr}
-					onClick={onSubmit}
-				>
+				<PassrLogo width="300px" height="300px" />
+
+				<TextInput
+					label="Email"
+					required
+					placeholder={'example@example.com'}
+					type="text"
+					value={email}
+					onChange={(e) => {
+						if (validEmail(e.target.value)) setEmailErr(false);
+						setEmail(e.target.value);
+					}}
+					onBlur={(e) => {
+						if (!validEmail(e.target.value)) setEmailErr(true);
+					}}
+				/>
+				<TextInput
+					label="Confirm Email"
+					required
+					placeholder="example@example.com"
+					type="text"
+					value={confirmEmail}
+					onChange={(e) => {
+						if (compareEmail(email, confirmEmail))
+							setEmailErr(false);
+						setConfirmEmail(e.target.value);
+					}}
+					onBlur={() => {
+						if (!compareEmail(email, confirmEmail))
+							setEmailErr(true);
+					}}
+				/>
+				<TextInput
+					label="Password"
+					required
+					type="password"
+					value={userPass}
+					onChange={(e) => {
+						if (validPass(e.target.value)) setPassErr(false);
+						setUserPass(e.target.value);
+					}}
+				/>
+
+				<TextInput
+					label="Confirm Password"
+					required
+					type="password"
+					value={confirmUserPass}
+					onChange={(e) => {
+						if (comparePass(userPass, confirmUserPass))
+							setPassErr(false);
+						setConfirmUserPass(e.target.value);
+					}}
+				/>
+
+				<SignUpButton disabled={emailErr || passErr} onClick={onSubmit}>
 					Sign Up
 				</SignUpButton>
+				<TextLinkContainer>
+					<TextLink
+						text="Already have an account? Sign in here."
+						onClick={() => {
+							history.push('/sign-in');
```

Comment:
Same question as Seiya; what is `useHistory` for?

Reply:
As stated in the response to Seiya

So you're definitely familiar with the browser history (the back and forward button). This essentially allows you to access this history through the react-router lib. When you push onto the history, it changes your page and updates the URL. It also allows the user to hit the browser back button to go to the last page they were on, giving the application a multi page feel, while actually being an SPA.